src/pathology-report/ReportComponent.tsx

```diff
@@ -1,40 +1,476 @@
+    },
+    {
+      key: "sampleStatus",
+      header: "Sample status",
+    },
+    {
+      key: "dateOfRequest",
+      header: "Date of Request",
+    },
+    {
+      key: "referralStatus",
+      header: "Referral status",
+    },
+    {
+      key: "sampleDropOff",
+      header: "Sample drop off?",
+    },
+  ];
 
   React.useEffect(() => {
-    getPatientEncounters('').then(setEncountersList);
+    getUserLocation().then(setUserLocation);
+    getLocations().then(setListOfHospitals);
+    getConceptAnswers(SAMPLESTATUSUUID).then(setSampleStatusResults);
+    getConceptAnswers(REFERRALSTATUSUUID).then(setReferralStatusResults);
+    getEncounters().then(setEncountersList);
   }, []);
 
+  const handleSampleStatus = (value) => {
+    setSampleStatus(value);
+  };
+  const handleReferralStatus = (value) => {
+    setReferralStatus(value);
+  };
+  const handleSendingHospital = (value) => {
+    setSendingHospital(value);
+  };
+  const handlePatientName = (value) => {
+    setpatientName(value);
+  };
+
+  const filteredEncList = encountersList
+    .filter(
+      (encList) =>
+        !sampleStatus ||
+        (encList.sampleStatusObs &&
+          encList.sampleStatusObs.toLowerCase() == sampleStatus.toLowerCase())
+    )
+    .filter(
+      (encList) =>
+        !referralStatus ||
+        (encList.referralStatusObs &&
+          encList.referralStatusObs.toLowerCase() ==
+            referralStatus.toLowerCase())
+    )
+    .filter(
+      (encList) =>
+        !sendingHospital ||
+        (encList.patientHealthCenter &&
+          encList.patientHealthCenter.toLowerCase() ==
+            sendingHospital.toLowerCase())
+    )
+    .filter(
+      (encList) =>
+        !patientName ||
+        (encList.family_name &&
+          encList.family_name
+            .toLowerCase()
+            .includes(patientName.toLowerCase())) ||
+        (encList.family_name2 &&
+          encList.family_name2
+            .toLowerCase()
+            .includes(patientName.toLowerCase())) ||
+        (encList.middle_name &&
+          encList.middle_name
+            .toLowerCase()
+            .includes(patientName.toLowerCase())) ||
+        (encList.given_name &&
+          encList.given_name.toLowerCase().includes(patientName.toLowerCase()))
+    );
+  // const rows = filteredEncList.map(
+  //   (encounterInfo) => {
+  //       return {
+  //           // patientLink: <a href={`/openmrs/patientDashboard.form?patientId=${encounterInfo.personId}`} >Link</a>,
+  //           ...
+
+  var rows = filteredEncList.map((encounterInfo) => {
+    return {
+      // rows.push({
+      // patientLink: <a href={`/openmrs/patientDashboard.form?patientId=${encounterInfo.personId}`} >Link</a>,
+      id: `${encounterInfo.encounterId}`,
+      patientNames: (
+        <a
+          href={`/openmrs/patientDashboard.form?patientId=${encounterInfo.personId}`}
+        >
+          {encounterInfo.family_name ? encounterInfo.family_name : ""}{" "}
```

Comment:
Did you try simply

```suggestion
          {encounterInfo.family_name}{" "}
```

? I think that should work, since React renders `null` as nothing (same as `""`).